scripts/service.py

```diff
@@ -0,0 +1,247 @@
+            self.logErr("execute script error:" + str(err))
+            
+    def _readRequest(self):
+        """ read all that is written in the pipe, so in case of input storm,
+            just do work once.
+        """
+        if self.fifo != None:
+            try:
+                while True:
+                    self.buffer = os.read(self.fifo, BUFFER_SIZE)
+                    if len(self.buffer) == 0:
+                        # hung up
+                        self.logErr("hung up found while read, exiting")
+                        raise OSError
+            except OSError as err:
+                if err.errno == errno.EAGAIN or err.errno == errno.EWOULDBLOCK:
+                    return True # assume already read as invoked after poll
+        else:
+            return False
+            
+    def _waitForInput(self):
+        """ hang on the poll to wait for incoming requests.
+            As we hold one writer, there should be no EPOLLHUP when others closed the pipe.
+        """
+        events = self.epollObj.poll(-1)
+        for fileno, event in events:
+            if fileno == self.fifo and event & select.EPOLLIN:
+                return True
+            elif fileno == self.fifo and event & (select.EPOLLERR | select.EPOLLHUP):
+                # probably some log and raise failure
+                self.logErr("exit due to unexpected events:" + str(event))
+                raise OSError
+
+        return False
+                
+    def work(self):
+        """ server side daemon logic """
+        self.logInfo("server started")
+        
+        try:
+            self._openRead()
+            while True:
+                if self._waitForInput():
+                    if self._readRequest():
+                        self._executeSub()
+        except Exception as err:
+            self.logErr("Exit at error:" + str(err))
+            self._closeFIFO()
+
+
+class ClientInstance(ServiceSkeleton):
+    """ client side implementation, to invoke the server/daemon to do work """
+    def __init__(self, name):
+        super(ClientInstance, self).__init__(name)
+        
+    def _writeRequest(self):
+        """ write the pipe to wake up the daemon to do the work """
+        if self.fifo == None:
+            self._openWrite()
+            
+        try:
+            os.write(self.fifo, self.name)
+        except OSError as err:
+            self.logErr("Write error at:" + str(err))
+            raise err
+
+    def _openWrite(self):
+        """ open pipe for write """
+        try:
+            self.fifo = os.open(self.pipePath, os.O_WRONLY)
+        except OSError as err:
+            self.logErr("open file write error:" + str(err))
+            raise err
+            
+    def work(self):
+        """ simply write the request and end ourselves """
+        self.logInfo("client started")
+        
+        try:
+            self._openWrite()
+            self._writeRequest()
+            self._closeFIFO()
```

Comment:
This could be moved to a finally block to avoid duplication with the error case.